src/serde/reader.rs

```diff
@@ -54,7 +54,7 @@ impl Error for DeserializeError {
             DeserializeError::IoError(ref err) => Error::description(err),
             DeserializeError::InvalidEncoding(ref ib) => ib.desc,
             DeserializeError::SizeLimit => "the size limit for decoding has been reached",
-            DeserializeError::Serde(ref s) => s.description(),
+            DeserializeError::Custom(_) => "a custom serialization error was reported",
```

Comment:
You should be able to return the real error here. `DeserializeError::Custom(ref msg) => msg`